source/IllegalHailPanel.cpp

```diff
@@ -0,0 +1,138 @@
+/* IllegalHailPanel.cpp
+Copyright (c) 2021 by quyykk
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "IllegalHailPanel.h"
+
+#include "text/alignment.hpp"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "text/Format.h"
+#include "GameData.h"
+#include "Government.h"
+#include "Information.h"
+#include "Interface.h"
+#include "Messages.h"
+#include "Planet.h"
+#include "PlayerInfo.h"
+#include "Politics.h"
+#include "Ship.h"
+#include "ShipEvent.h"
+#include "Sprite.h"
+#include "UI.h"
+
+#include <algorithm>
+#include <cmath>
+
+using namespace std;
+
+
+
+IllegalHailPanel::IllegalHailPanel(PlayerInfo &player, const Ship &hailingShip, Ship &scannedShip,
+		const Politics::Punishment &fine)
+	: player(player), hailingShip(hailingShip), scannedShip(scannedShip), fine(fine)
+{
+	const Government *gov = hailingShip.GetGovernment();
+	if(!hailingShip.Name().empty())
+		header = gov->GetName() + " " + hailingShip.Noun() + " \"" + hailingShip.Name() + "\"";
+	else
+		header = hailingShip.DisplayModelName() + " (" + gov->GetName() + ")";
+	header += " is hailing you:";
+
+	static const std::string defaultMessage
+		= "You've been detected carrying illegal <type> and have been issued a fine of <fine>."
+			" \n\tDump your cargo immediately or we'll be forced to disable and board your ship.";
```

Comment:
```suggestion
			" \n\tDump your cargo immediately or we'll be forced to open fire.";
```
seems better as a default to me?